packages/fx-core/src/core/middleware/projectMigratorV3.ts

```diff
@@ -421,3 +440,51 @@ export async function userdataMigration(context: MigrationContext): Promise<void
       }
   }
 }
+
+export async function generateApimPluginEnvContent(context: MigrationContext): Promise<void> {
+  // general
+  if (await context.fsPathExists(path.join(".fx", "configs", "projectSettings.json"))) {
+    const projectSettingsContent = fs.readJsonSync(
+      path.join(context.projectPath, ".fx", "configs", "projectSettings.json")
+    );
+    // judge if apim plugin exists
+    let flag_apimPlugin = false;
+    for (const obj of projectSettingsContent["components"])
+      if (obj["name"] === "apim") {
+        flag_apimPlugin = true;
+        break;
+      }
+    // must do this step after configs migration
```

Comment:
why we must do this step after configs migration?

Reply:
I think it is better to add apim plugin information into .env after .env files already exist. Actually, there is no need to have to do this after configs migration